src/control/cmd/daos_agent/mgmt_rpc_test.go

```diff
@@ -317,6 +283,48 @@ func TestAgent_mgmtModule_getAttachInfo(t *testing.T) {
 			},
 			expResp: &mgmtpb.GetAttachInfoResp{Status: int32(daos.Unreachable)},
 		},
+		"client telemetry enabled": {
+			reqBytes: reqBytes(&mgmtpb.GetAttachInfoReq{}),
+			tmCfg: &TelemetryConfig{
+				Port:    1234,
+				Enabled: true,
+			},
+			expResp: func(resp *mgmtpb.GetAttachInfoResp) *mgmtpb.GetAttachInfoResp {
+				resp.ClientNetHint.EnvVars = append(resp.ClientNetHint.EnvVars,
+					fmt.Sprintf("%s=1", telemetry.ClientMetricsEnabledEnv),
+					fmt.Sprintf("%s=1", telemetry.ClientMetricsRegisterEnv),
+				)
+				return resp
+			}(respWith(testResp, "test1", "dev1", defNumaMap)),
+		},
+		"client telemetry enabled; self not in register pattern": {
```

Comment:
How about the case where the self is in the register pattern?

Reply:
Done.